controllers/validation.js

```diff
@@ -0,0 +1,626 @@
+ * @param  {Array} - Array of ValidationErrors from validation
+ * @param  {Array} - Array to store processed ErrorObjs in
+ */
+function processErrors(errors, processedErrors, schema){
+	const length = errors.length;
+	let counter;
+	for (counter = 0; counter < length; counter++){
+
+		switch (errors[counter].name){
+		case 'required':
+			handleMissingError(processedErrors, errors, counter, schema);
+			break;
+		case 'type':
+			handleTypeError(processedErrors, errors, counter);
+			break;
+		case 'format':
+		case 'pattern':
+			handleFormatError(processedErrors, errors, counter);
+			break;
+		case 'enum':
+			handleEnumError(processedErrors, errors, counter);
+			break;
+		case 'dependencies':
+			handleDependencyError(processedErrors, errors, counter);
+			break;
+		case 'anyOf':
+			handleAnyOfError(processedErrors, errors, counter);
+			break;
+		}
+	}
+}
+
+function makeFieldReadable(input){
+
+	return input
+	.replace(/([A-Z])/g, ' $1')
+	.replace(/^./, function(str){
+		return str.toUpperCase();
+	})
+	.replace('Z I P', 'Zip')
+	.replace('U R L', 'URL');
+
+}
+
+function makePathReadable(input){
+
+	if (typeof input === 'string'){
+		const parts = input.split('.');
+		const readableParts = [];
+		let readablePath = '';
+		parts.forEach((field)=>{
+			readableParts.push(makeFieldReadable(field));
+		});
+		readablePath = readableParts.shift();
+		readableParts.forEach((part)=>{
+			readablePath = `${readablePath}/${part}`;
+		});
+		return readablePath;
+	}
+	else {
+		return false;
+	}
+
+}
+
+function buildFormatErrorMessage(fullPath){
+	const field = fullPath.substring(fullPath.lastIndexOf('.') + 1);
+	const readablePath = makePathReadable(fullPath);
+	const errorMessage = `${readablePath}${errors[field]}`;
+	return errorMessage;
+
+}
+
+/**
+ * Creates error message for anyOf errors
+ * 
+ * @param  {array} anyOfFields - list of fields, at least one being required.
+ * @return {string}
+ */
+function makeAnyOfMessage(anyOfFields){
+	if (anyOfFields){
+		let output, count = 1;
+		const length = anyOfFields.length;
+		output = `${makePathReadable(anyOfFields[0])}`;
+		while (count < length) {
+			const field = anyOfFields[count];
+			output = `${output} or ${makePathReadable(field)}`;
+			count ++;
+		}
+		return output;
+	}
+	else {
+		return false;
+	}
+}
+
+function concatErrors(errorMessages){
+
+	let output = '';
+	errorMessages.forEach((message)=>{
+		output = `${output}${message} `;
+	});
+	output = output.trim();
+	return output;
+}
+
+function generateFileErrors(output, error, messages){
+	const reqFile = `${makePathReadable(error.field)} is a required file.`;
+	const small = `${makePathReadable(error.field)} cannot be an empty file.`;
+	const large = `${makePathReadable(error.field)} cannot be larger than ${error.expectedFieldType} MB.`;
+	let invExt, invMime;
+	if (typeof(error.expectedFieldType) !== 'undefined' && error.expectedFieldType.constructor === Array){
```

Comment:
expectedFieldType is a string for field errors. Should either use a different field to pass this information or make this difference clear somewhere.